### PR TITLE
feat[new-user]: api implemented

### PR DESCRIPTION
**Pull request by DNALOCK**

### CODE REVIEW

File: usecase/user_admin_usecase.go [hunk 1]
```diff
@@ -28,6 +33,47 @@ func NewUserAdminUsecase(
...
+		Roles:           []model.Role{role},
+	}
+	err := uiu.userRepository.CreateFirstUser(user)
+	if err != nil {
+		logrus.Errorf("error creating the first user: %v", err.Error())
+		return model.User{}, err
+	}
+	return user, nil
+}
+
```
Comment (StevenDB995) [inline usecase/user_admin_usecase.go:76]: Please put them in a new go package, say new_user_init_usecase.go

**Review by StevenDB995 — CHANGES_REQUESTED** [review]

File: usecase/user_admin_usecase.go [hunk 2]
```diff
@@ -28,6 +33,47 @@ func NewUserAdminUsecase(
...
 
+func NewUserInitUsecase(
+	userRepository model.UserRepository,
+) model.NewUserInitUsecase {
+	return &userInitUsecase{
+		userRepository: userRepository,
+	}
+}
+
+func (uiu *userInitUsecase) CreateFirstUser() (model.User, error) {
```
Comment (StevenDB995) [inline usecase/user_admin_usecase.go:44]: Please put them in a new go package, say new_user_init_usecase.go

File: usecase/user_admin_usecase.go [hunk 3]
```diff
@@ -28,6 +33,47 @@ func NewUserAdminUsecase(
...
+	}
+	if count != 0 {
+		logrus.Infof("error counting number of users")
+		return model.User{}, errors.New("has users")
+	}
+	logrus.Infof("error counting number of users: %d", count)
+	role := model.Role{
+		ID:   0,
+		Name: "Root Administrator",
+	}
```
Comment (StevenDB995) [inline usecase/user_admin_usecase.go:58]: You cannot create a role with ID specified unless you modified the database schema. For the "id" column of the "roles" table, it is currently "GENERATED ALWAYS AS IDENTITY", and you have to change it to "GENERATED BY DEFAULT AS IDENTITY"

**Review by StevenDB995 — APPROVED** [review]

**Review by edwardlin1695 — APPROVED** [review]
Changes to aforementioned comments made by Steven look good.
I assume unit testing has been conducted but currently do not see any related documentation

**Pull request merged**